infrastructure/readme.md

```diff
@@ -1,247 +1,307 @@
-7. On the **Users** section add **Anonymous** user
-8. Give **Deploy/Cache, Annotate, Read** permissions
-9. Click **Save & Finish**
-
-## Managing the Docker services
-The services are managed with docker-compose.
-
-Initial startup is trigered from the installation script with the command: ```docker-compose up -d```.
-This will kickstart the creation of the containers and will boot them up.
-
-After that you can use the following management commands (executed from the directory of the docker-compose file):
-* ```docker-compose start``` - will start all containers for all services in the docker-compose file
-* ```docker-compose stop``` - will stop all containers for all services in the docker-compose file
-* ```docker-compose restart```- will restart all containers for all services in the docker-compose file
-
-* ```docker-compose up -d``` - this will recreate all Docker containers
-* ```docker-compose down``` - this will stop all containers and delete them
-
-## Uninstalling the solution
-To uninstall the solution run the following command:
-```../infrastructure/iac-for-vrealize-configurator.sh --clean```
-
-This will remove:
-* All running containers for the solution - NGINX, GitLab, GitLab CI Runner, Artifactory
-* Docker network **infranet**
-* JFrog CLI
-* Custom configuration from **/etc/hosts** file
-
-No persistent container data is deleted - if you want to fully delete all data you should
-delete the files located in the following locations:
-
-NGINX:
-* ./nginx/conf.d
-* ./nginx/vhost.d
-* ./nginx/html
-* ./nginx/certs
-* ./nginx/proxy.conf
-* /var/log/nginx
-
-GitLab:
-* /srv/gitlab/config
-* /srv/gitlab/logs
-* /srv/gitlab/data
-
-GitLab Runner:
-* /srv/gitlab-runner/config
-
-Artifactory:
-* /data/artifactory
-
-# Next step
-- Configure developer **[Build Tools for VMware Aria Workstation](./setup-workstation.md)**
+7. Configure the GitLab pipeline by creating a new file in the root of the repository named `.gitlab-ci.yml`:
+    ```
+    stages:
+      - setup
+      - build
+      - test
+      - install
+
+    variables:
+      BUILD_OPTS: >-
+        -Dhttps.protocols=TLSv1.2
+        -Dmaven.repo.local=$CI_PROJECT_DIR/.m2/repository
+        -DskipTests=true
+      DEPLOY_OPTS: >-
+        -Dhttps.protocols=TLSv1.2
+        -Dmaven.repo.local=$CI_PROJECT_DIR/.m2/repository
+        -DskipTests=true
+        -Dbuild.number=$CI_PIPELINE_IID
+        -Dsurefire.useSystemClassLoader=false
+        -Pbundle-with-installer
+        -DoutputDirectory=target
+        -DartifactName=artifact.zip
+        -U
+        --batch-mode
+
+    dynamic_variables:
+      stage: setup
+      script:
+        - echo "GROUP_ID=$(mvn help:evaluate -Dexpression=project.groupId -q -DforceStdout)" >> build.env
+        - echo "ARTIFACT_ID=$(mvn help:evaluate -Dexpression=project.artifactId -q -DforceStdout)" >> build.env
+        - echo "PROJECT_VERSION=$(mvn help:evaluate -Dexpression=project.version -q -DforceStdout)" >> build.env
+      artifacts:
```

Comment:
For some reason the `artifacts` part fails to execute
```
WARNING: Uploading artifacts as "dotenv" to coordinator... POST http://infra.corp.local/gitlab/api/v4/jobs/1/artifacts: 400 Bad Request (Invalid Format)  id=1 responseStatus=400 Bad Request status=400 token=glcbt-64
FATAL: invalid argument                            
ERROR: Job failed: exit status 1
```
Commenting this out makes the build successful